ext/ffi/ffi.c

```diff
@@ -911,12 +917,88 @@ static void zend_ffi_cdata_set(zval *object, zval *value) /* {{{ */
 	}
 #endif
 
+	if (UNEXPECTED(Z_TYPE_P(member) != IS_STRING)
+	 || UNEXPECTED(!zend_string_equals_literal(Z_STR_P(member), "cdata"))) {
+		zend_throw_error(zend_ffi_exception_ce, "only 'cdata' property may be set");
+		return;
+	}
+
 	zend_ffi_zval_to_cdata(cdata->ptr, type, value);
+
+	return value;
 }
 /* }}} */
 
 static int zend_ffi_cdata_cast_object(zval *readobj, zval *writeobj, int type) /* {{{ */
 {
+	if (type == IS_STRING) {
+		zend_ffi_cdata *cdata = (zend_ffi_cdata*)Z_OBJ_P(readobj);
+		zend_ffi_type  *type = ZEND_FFI_TYPE(cdata->type);
```

Comment:
The shadowing of the `type` argument is a bit confusing here.

Generally, would it be possible to reuse the code from zend_ffi_cdata_to_zval?

Reply:
"type" renamed to "ctype".

Usage of zend_ffi_cdata_to_zval() dosn't make sense because it contains wrong paths and inlined anyway.